saksbehandler/server/src/routes.ts

```diff
@@ -1,56 +1,55 @@
+import labsProxy from '@/proxy/labs-proxy';
+import decoratorProxy from '@/proxy/decorator-proxy';
+import { isTokenValid } from '@/auth/token';
 
 const router = express.Router();
 
-const ensureAuthenticated =
-    (azureJwksClient: JwksClient) => async (req: Request, res: Response, next: NextFunction) => {
-        if (req.headers['authorization']) {
-            const bearerToken = req.headers.authorization?.replace('Bearer', '').trim();
-            const decoded = jwt.decode(bearerToken, { complete: true });
-            if (decoded) {
-                const verifyOptions: jwt.VerifyOptions = {
-                    algorithms: ['RS256'],
-                };
-                const key = await azureJwksClient.getSigningKey(decoded.header.kid);
-                const signingKey = key.getPublicKey();
-                jwt.verify(bearerToken, signingKey, verifyOptions, (error) => {
-                    if (!error) {
-                        next();
-                    } else {
-                        res.redirect('/login');
-                    }
-                });
+export default () => {
+    // Unprotected
+    router.get('/isAlive', (_, res) => {
+        res.status(200).send('Alive');
+    });
+    router.get('/isReady', (_, res) => {
+        res.status(200).send('Ready');
+    });
+
+    if (process.env.MILJO === 'dev-gcp-labs') {
+        router.use('/logout', (_, res) => {
+            res.clearCookie('aad-token');
+            res.redirect('/');
+        });
+
+        labsProxy(router);
+    } else {
+        router.get('/login', (_, res) => {
+            res.redirect('/oauth2/login');
+        });
+        router.get('/logout', (_, res) => {
+            res.redirect('/oauth2/logout');
+        });
+
+        router.use(async (req, res, next) => {
+            if (!req.headers.authorization) {
+                res.sendStatus(401);
+            } else if (await isTokenValid(req)) {
+                next();
             } else {
                 res.redirect('/login');
             }
-        } else {
-            res.redirect('/login');
-        }
-    };
-
-export default (authClient: Client, tokenEndpoint: TokenEndpoint, azureJwksClient: JwksClient) => {
-    // Unprotected
-    router.get('/isAlive', (req, res) => res.send('Alive'));
-    router.get('/isReady', (req, res) => res.send('Ready'));
-
-    router.get('/login', (req, res) => res.redirect('/oauth2/login'));
-    router.get('/logout', (req, res) => res.redirect('oauth2/logout'));
+        });
 
-    router.use(ensureAuthenticated(azureJwksClient));
+        apiProxy(router);
+    }
 
-    apiProxy(router, authClient, tokenEndpoint);
-    decoratorProxy(router, authClient, tokenEndpoint);
+    decoratorProxy(router, process.env.MILJO === 'dev-gcp-labs');
 
-    // serve static files
     router.use(express.static(path.join(__dirname, '../build')));
 
-    router.get('/*', (req, res) => {
+    router.get('/{*splat}', (req, res) => {
```

Comment:
Hvorfor gjør vi dette? Prøvde å skjønne hva splat-argumentet gjør men klarte ikke helt å få tak i det.

Reply:
> The wildcard * must have a name, matching the behavior of parameters :, use /*splat instead of /*

~ https://expressjs.com/en/guide/migrating-5.html